powersimdata/network/europe_tub/model.py

```diff
@@ -0,0 +1,140 @@
+    :param bool overwrite: the existing dataset is deleted a new dataset is downloaded
+        from zenodo.
+    """
+
+    def __init__(self, interconnect, reduction=None, overwrite=False):
+        """Constructor."""
+        self.grid_model = "europe_tub"
+        self.interconnect = check_and_format_interconnect(
+            interconnect, model=self.grid_model
+        )
+        self.data_loc = os.path.join(os.path.dirname(__file__), "data")
+        self.zenodo_record_id = "3601881"
+        self.reduction = reduction
+
+        if overwrite:
+            self.remove_data()
+
+        self.retrieve_data()
+
+    def remove_data(self):
+        """Remove data stored on disk"""
+        print("Removing PyPSA-Eur dataset")
+        shutil.rmtree(self.data_loc)
+
+    def retrieve_data(self):
+        """Fetch data"""
+        zenodo_get.zenodo_get([self.zenodo_record_id, "-o", f"{self.data_loc}"])
+        with ZipFile(os.path.join(self.data_loc, "networks.zip"), "r") as zip_network:
+            zip_network.extractall(self.data_loc)
+
+    def build(self):
+        """Build network"""
+        path = os.path.join(self.data_loc, "networks")
+        if self.reduction is None:
+            network = pypsa.Network(os.path.join(path, "elec_s.nc"))
+        elif os.path.exists(os.path.join(path, f"elec_s_{self.reduction}.nc")):
+            network = pypsa.Network(os.path.join(path, f"elec_s_{self.reduction}.nc"))
+        else:
+            raise ValueError(
+                "Invalid Resolution. Choose among: None | 1024 | 512 | 256 | 128 | 37"
+            )
+        id2zone = {i: l for i, l in enumerate(network.buses.index)}
+        zone2id = {l: i for i, l in id2zone.items()}
+
+        if "Europe" in self.interconnect:
+            self.network = network
+            self.id2zone = id2zone
+            self.zone2id = zone2id
+        else:
+            filter = list(  # noqa: F841
+                interconnect2abv[
+                    interconnect_to_name(self.interconnect, model=self.grid_model)
+                ]
+            )
+            self.network = network[network.buses.query("country == @filter").index]
+            self.zone2id = {l: zone2id[l] for l in self.network.buses.index}
+            self.id2zone = {i: l for l, i in self.zone2id.items()}
+
+        self.model_immutables = self._generate_model_immutables()
+
+    def _generate_model_immutables(self):
+        """Generate the model immutables"""
+        mapping = ModelImmutables(self.grid_model, interconnect=self.interconnect)
+
+        # loadzone
+        mapping.zones["loadzone"] = set(self.zone2id)
+        mapping.zones["id2loadzone"] = self.id2zone
+        mapping.zones["loadzone2id"] = self.zone2id
+        mapping.zones["loadzone2abv"] = self.network.buses["country"].to_dict()
+        mapping.zones["loadzone2country"] = {
+            l: abv2country[a]
+            for l, a in self.network.buses["country"].to_dict().items()
+        }
```

Comment:
Is this equivalent to `mapping.zones["loadzone2country"] = self.network.buses["country"].map(abv2country).to_dict()`?

Reply:
Done